xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -309,8 +309,8 @@ func (b *cdsBalancer) UpdateClientConnState(state balancer.ClientConnState) erro
 	b.lbCfg = lbCfg
 
 	// Handle the update in a blocking fashion.
-	done := make(chan struct{})
-	ok = b.serializer.Schedule(func(context.Context) {
+	errCh := make(chan error, 1)
```

Comment:
Similar here as above.

Reply:
I can't make it unbuffered here because if `ScheduleOr` fails, it will call `onFailure` inline and this means that the write to the channel on line 327 `errCh <- errBalancerClosed` will happen before the read on line 330. Therefore it will just hang.

Reply:
Ah yes, that makes sense.